app/views.py

```diff
@@ -78,6 +78,106 @@ def admin():
     return render_template("admin.html", vehicles=vehicles, services=services)
 
 
+@app.route("/admin/services")
+def admin_services_management():
+    services = services_model.all()
+    return render_template("admin/services/index.html", services=services)
+
+
+@app.route("/admin/services/save", methods=["POST"])
+def admin_services_save():
+    updated_images = None
+    if request.files.getlist("images"):
+        updated_images = util.upload_files(
+            request.files.getlist("images"), app.config["IMAGES_DIR_PATH"]
+        )
+    price = {
+        price[0]: price[1]
+        for price in list(
+            zip(
+                request.form.getlist("price_names[]"),
+                request.form.getlist("price_values[]"),
+            )
+        )
+    }
+    try:
+        service = services_model.get(id=request.form.get("service_id"))
+        # get current set of images
+        images = set(service["images"])
+        # add updated images if any
+        images |= set(updated_images)
```

Comment:
hmmmm first time seeing this `|=` can you explain what does it do?

Reply:
the `|=` is union operation of set data type. The above meaning that I take current list of images of the service union with the new uploaded images. I'm using set to avoid duplicated image names

Reply:
nice! I did not know that! thank you!